### PR TITLE
Tokenizer test cases and reformatting of tokenizer training file

### PR DESCRIPTION
**Pull request by chandanms**
## Description
Added test cases for tokenizer to verify the functionality of test cases both after training and after pruning. Reformatted the tests to use `create_tokenizer` function directly to make it more aligned with the creation script. 

Removed COMMON_PREFIXES and COMMON_SUFFIXES from initial_alphabet as empirical testing (trained with and without it) showed the WordPiece trainer naturally discovers these frequent morphemes during training, making explicit seeding redundant and causing unnecessary complexity.  We should have validated this assumption earlier rather than adding potentially redundant configuration.

## Related Issue
Sanity check on using special tokens correctly

## Motivation and Context
The training and pruning tokenizer could potentially cause unwanted risk which would be deeply problematic if not discovered. Hence the thorough checking of edge cases and use of special tokens.

## How Has This Been Tested?
Added tests to check the behaviour and reformatted the tests to test the functionality after training and after pruning.

## Does this PR introduce a breaking change?
No

### CODE REVIEW

**Comment by chandanms** [comment]
Check only the last commit please. I had already started coding and didnt see the previous merge for #40

File: tests/test_tokenizer.py [hunk 1]
```diff
@@ -2,73 +2,107 @@
...
 def create_test_data_iterator(test_data: list[str]) -> Generator[str, None, None]:
     """Create iterator from test data."""
     yield from test_data
 
 
+def create_test_tokenizer():
+    """Create a fresh tokenizer for testing."""
+    train_data = ["hello world", "hello there", "world peace", "simple stories"]
+    train_iter = create_test_data_iterator(train_data)
+    return train_tokenizer(train_iter, vocab_size=200)
```
Comment (danbraunai) [inline tests/test_tokenizer.py]: You should be able to just do `return train_tokenizer(iter(train_iter), vocab_size=200` and remove the create_test_data_iterator function.

File: tests/test_tokenizer.py [hunk 2]
```diff
@@ -2,73 +2,107 @@
...
+    assert "[UNK]" in vocab_orig and "[EOS]" in vocab_orig
 
-    assert "[UNK]" in vocab and "[EOS]" in vocab
-    assert vocab["[UNK]"] in [0, 1] and vocab["[EOS]"] in [0, 1]
+    test_data = ["hello world"]
+    data_iter = create_test_data_iterator(test_data)
+    pruned = prune_tokenizer(data_iter, tokenizer)
+    vocab_pruned = pruned.get_vocab()
+    assert "[UNK]" in vocab_pruned and "[EOS]" in vocab_pruned
+    assert vocab_pruned["[UNK]"] in [0, 1] and vocab_pruned["[EOS]"] in [0, 1]
```
Comment (danbraunai) [inline tests/test_tokenizer.py:25]: Maybe assert that they're not both 0 or 1 too. (i.e. that they're not equal).

**Review by danbraunai — APPROVED** [review]
Nice. Minor comments below. I love the removal of COMMON_PREFIXES and COMMON_SUFFIXES. Best also to add that this is done to the README, because it differs to the paper (bottom of page 5 https://arxiv.org/pdf/2504.09184).

Can merge after addressing those.
